src/api/handlers.rs

```diff
@@ -0,0 +1,83 @@
+use std::sync::Arc;
+use axum::http::StatusCode;
+use axum::extract::{Json, State};
+
+use super::errors::AppError;
+
+use crate::state::AppState;
+use crate::storage::local::LocalStorage;
+use crate::track::Track;
+
+pub mod tracks {
+    use crate::request::{TrackRequest, SearchTrackRequest};
+    use super::*;
+
+    pub async fn add(
+        State(state): State<Arc<AppState<LocalStorage>>>,
+        Json(track_request): Json<TrackRequest>
+    ) -> Result<Json<Track>, AppError> {
+        let track_id = track_request.track_id.clone();        
+        tracing::info!("received a request to add a new track to a playlist - {}", track_id); 
+        match state.api_client.get_by_id(&track_id).await {
+            Ok(track) => {
+                tracing::info!("{:?}", track);
+                if let Err(err) = state.interface.add_track(track.clone()).await {
+                    Err(AppError::InternalServerError(err))
+                } else {
+                    Ok(Json(track))
+                }
+            },
+            Err(err) => {
+                Err(AppError::InternalServerError(err))
+            } 
+        }
```

Comment:
as I understand, it can be simplified:

```suggestion
        let track = state.api_client.get_by_id(&track_id).await?;
        state.interface.add_track(track.clone()).await?;
        Ok(track)
```

Reply:
Thank you! Fixed. I made a small change to your solution - the method should return `Ok(Json(track))` instead of `Ok(track)`.